src/server/project.ts

```diff
@@ -330,6 +357,70 @@ namespace ts.server {
             }
         }
 
+        getReferencedFiles(filename: string): string[] {
+            if (!this.languageServiceEnabled) {
+                return [];
+            }
+
+            const sourceFile = this.program.getSourceFile(filename);
+            const modules = sourceFile.resolvedModules;
+            const result: string[] = [];
+            if (modules) {
+                // We need to use a set here since the code can contain the same import twice,
+                // but that will only be one dependency.
+                const referencedModules: Map<boolean> = {};

```

Comment:
technically we should also include type reference directives into the list of dependencies